cores/esp8266/umm_malloc/umm_malloc.cpp

```diff
@@ -889,6 +975,29 @@ void *umm_malloc(size_t size) {
         return ptr;
     }
 
+#if UMM_ENABLE_MEMALIGN
+
+    /*
+     * Ensure alignment is power of 2; however, we allow zero.
+     */
+
+    if (0u != (alignment & (alignment - 1u))) {
+      STATS__ALIGNMENT_ERROR(id_malloc, alignment);
+      return ptr;
+    }
+
+    /*
+     * Allocations default to 8-byte alignment same as __STDCPP_DEFAULT_NEW_ALIGNMENT__
+     * No need to execute extra alignment logic for small alignments.
+     */
+
+    static_assert(__STDCPP_DEFAULT_NEW_ALIGNMENT__ == 8u);
+    static_assert(__STDCPP_DEFAULT_NEW_ALIGNMENT__ == sizeof(umm_block));
+    if (alignment <= sizeof(umm_block)) {
+        alignment = 0u;   // Use implementation default, 8.
+    }
+#endif
```

Comment:
When I add `-faligned-new=4` or other non-8 values to the build the GCC compiler complains with:
```
cc1: warning: command-line option '-faligned-new=4' is valid for C++/ObjC++ but not for C
```
It will also complain with 16. 

I will remove the `static_asserts`. That will allow the use of `-faligned-new=`


umm_malloc allocates in blocks of 8 bytes. 

Assume the total Heap range is in multiples of 8 bytes and starts at an 8-byte aligned address boundary. 
It costs a total of 8 bytes of space  (4 bytes at each end of the address range) for resulting data allocation addresses to be 8-byte aligned.

The umm_block structure is sized as two 4-byte fields. 
The first holds two 16-bit block addresses to the previous and next umm_block. 
The 2nd 4-byte field holds two 16-bit block addresses for the previous and next free umm_blocks. 
Take the case of a minimum-sized allocation ocuping 1 umm_block, 8-bytes. 
The 2nd 4-byte field becomes the start of the data address when allocated.

Once a block is allocated, the Heap overhead is 4 bytes leaving 4 bytes for data space. If we make sure the heap starts at an address that is 4-byte aligned but not 8-byte aligned, then the resulting data address will be 8-byte aligned.  And, this will hold for all allocations.